virttest/utils_misc.py

```diff
@@ -1748,6 +1748,25 @@ def get_node_distance(self, node_id):
 
         return node_distance.strip().split()
 
+    def get_all_node_meminfo(self):
+        """
+        Get the complete meminfo of all nodes.
+
+        :return: List of node meminfo dict
+        :rtype: list
+        """
+        meminfo = []
+        meminfo_file = os.path.join(self.numa_sys_path, "node%s/meminfo")
+        for node in get_all_nodes():
+            node_meminfo = {}
+            meminfo_f = open(meminfo_file % node, 'r')
+            for info in meminfo_f.readlines():
+                key, value = re.match(r'Node \d+ (\S+):\s+(\d+)', info).groups()
+                node_meminfo[key] = value
+            meminfo_f.close()
+            meminfo.append(node_meminfo)
+        return meminfo
```

Comment:
I will be good to have `meminfo` as a dict with key as `node` instead of list, as we can't rely on the order in which list gets appended